src/network/netdev/macsec.c

```diff
@@ -0,0 +1,888 @@
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Failed to parse port for secure channel identifier. Ignoring assignment: %s",
+                           rvalue);
+                return 0;
+        }
+
+        *dest = htobe16(port);
+
+        TAKE_PTR(b);
+        TAKE_PTR(c);
+
+        return 0;
+}
+
+int config_parse_macsec_hw_address(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(macsec_receive_association_free_or_set_invalidp) ReceiveAssociation *b = NULL;
+        _cleanup_(macsec_receive_channel_free_or_set_invalidp) ReceiveChannel *c = NULL;
+        MACsec *s = userdata;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (streq(section, "MACsecReceiveChannel"))
+                r = macsec_receive_channel_new_static(s, filename, section_line, &c);
+        else
+                r = macsec_receive_association_new_static(s, filename, section_line, &b);
+        if (r < 0)
+                return r;
+
+        r = ether_addr_from_string(rvalue, b ? &b->sci.mac : &c->sci.mac);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Failed to parse MAC address for secure channel identifier. "
+                           "Ignoring assignment: %s", rvalue);
+                return 0;
+        }
+
+        TAKE_PTR(b);
+        TAKE_PTR(c);
+
+        return 0;
+}
+
+int config_parse_macsec_packet_number(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(macsec_transmit_association_free_or_set_invalidp) TransmitAssociation *a = NULL;
+        _cleanup_(macsec_receive_association_free_or_set_invalidp) ReceiveAssociation *b = NULL;
+        MACsec *s = userdata;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (streq(section, "MACsecTransmitAssociation"))
+                r = macsec_transmit_association_new_static(s, filename, section_line, &a);
+        else
+                r = macsec_receive_association_new_static(s, filename, section_line, &b);
+        if (r < 0)
+                return r;
+
+        r = safe_atou32(rvalue, a ? &a->sa.packet_number : &b->sa.packet_number);
+        if (r < 0) {
```

Comment:
according to the man page addition further up zero is not an OK value here, this should hence be filztered out?

Reply:
`0` means `unset`. So, usually setting `0` is meaningless, but not invalid. I'd like to keep the current logic.

Reply:
It's a mandatory value. Hence we should warn and verify in the verify function . Kernel will not accept this. See https://elixir.bootlin.com/linux/latest/source/drivers/net/macsec.c#L1645

Reply:
No no. When netlink message has the attribute, then it must be positive. But `networkd` does not send the attribute when packet number is `0`. So, it is not necessary to check the value.

Reply:
But kernel will fail to create since it verifies ``'MACSEC_SA_ATTR_PN``` is available . it not it fails there and if available then is this zero.
```
macsec-test: MACsec could not be add transmit secure association configurations: Invalid argument
````

Reply:
Ah, I notice that PN is mandatory for TXSA, but not for RXSA. I will add a check and update the man page. Thank you.

Reply:
Updated.